openedx_events/bridge/avro_attrs_bridge.py

```diff
@@ -123,131 +142,72 @@ def _record_fields_for_attrs_class(
-            # The default value is always set to None to allow attr class to handle dealing with default values
-            # in dict_to_attrs function in this class
-            if attribute.default is not attr.NOTHING:
-                inner_field["type"] = ["null", inner_field["type"]]
-                inner_field["default"] = None
-            field["type"]["fields"].append(inner_field)
-
+            field["type"]["fields"].append(
+                self._get_object_fields(attribute.name, attribute.type)
+            )
         return field
 
-    def to_dict(self, obj, event_overrides=None):
+    def to_dict(self, data_dict):
         """
-        Convert obj into dictionary that matches avro schema (self.schema).
+        Convert data_dict into dictionary that matches Avro schema (self.schema).
+
+        Warning: this does not validate that the data_dict input matches self._signal
 
         Args:
-            obj: instance of self._attr_cls
-            event_overrides: dict with following value overwrites:
-                - id: unique id for this event. If id is not in dict, a uuid1 will be created for this event
-                      For more info, see OEP-41: Asynchronous Server Event Message Format
-                - time: time stamp for this event. If time is not in dict, datetime.now() will be called
-                        For more info, see OEP-41: Asynchronous Server Event Message Format
+            data_dict: dict with all the values specified in OpenEdxPublicSignal.init_data.
         """
-        if isinstance(event_overrides, dict) and "id" in event_overrides:
-            event_id = event_overrides["id"]
-        else:
-            event_id = str(uuid.uuid1())
-        if isinstance(event_overrides, dict) and "time" in event_overrides:
-            event_timestamp = event_overrides["time"]
-        else:
-            event_timestamp = datetime.now().isoformat()
-        obj_as_dict = attr.asdict(obj, value_serializer=self._extension_serializer)
-        # Not sure if it makes sense to keep version info here since the schema registry will actually
-        # keep track of versions and the topic can have only one associated schema at a time.
-        avro_record = dict(
-            id=event_id,
-            type=self.config["type"],
-            time=event_timestamp,
-            source=self.config["source"],
-            sourcehost=self.config["sourcehost"],
-            minorversion=0,
-            data=obj_as_dict,
+        # TODO (EventBus): is there better way to do this besides using json?
+        # This first converts data_dict to json string and then back to dict.
+
+        return json.loads(
+            json.dumps(
+                data_dict, sort_keys=True, default=self._extension_signal_serializer
+            )
         )
-        return avro_record
 
-    def serialize(self, obj) -> bytes:
+    def _extension_signal_serializer(self, value):
```

Comment:
Wishing something in the docstring could better explain the difference or relationship between `_extension_signal_serializer` vs `_extension_serializer`.